### PR TITLE
Drop Python 3.6 support

### PR DESCRIPTION
**Pull request by seisman**
**Description of proposed changes**

- Remove Python 3.6 from our CI jobs
- Bump the minimum required Python version to 3.7 in the documentation
- Remove Python 3.6 from setup.py

Address #690.

**Reminders**

- [x] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

**Notes**

- You can write `/format` in the first line of a comment to lint the code automatically

### CODE REVIEW

**Comment by seisman** [comment]
Do we all agree that the next release will be v0.3.0? If so I will change the projects and milestone.

File: .github/workflows/ci_tests.yaml [hunk 1]
```diff
@@ -42,19 +42,10 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        python-version: [3.6, 3.7, 3.8, 3.9]
+        python-version: [3.7, 3.8, 3.9]
         os: [ubuntu-latest, macOS-latest, windows-latest]
-        # Run Python 3.6 tests on master branch only to conserve CI resources
         isMaster:
           - ${{ contains(github.ref, 'master') }}
```
Comment (weiji14) [inline .github/workflows/ci_tests.yaml:48]: These two lines can be removed too.

```suggestion
```

File: doc/install.rst [hunk 2]
```diff
@@ -16,7 +16,8 @@ Installing
 Which Python?
 -------------
 
-You'll need **Python 3.6 or greater** to run PyGMT.
+You'll need **Python 3.7 or greater** to run PyGMT. Old Python versions may
+work, but there is no guranteen that PyGMT will work as expected with old versions.
```
Comment (weiji14) [inline doc/install.rst:20]:
```suggestion
work, but there is no guarantee that PyGMT will work as expected with old versions.
```

**Review by weiji14 — COMMENTED** [review]
Just two minor things, and yes :+1: for going with v0.3.0.

**Review by weiji14 — APPROVED** [review]

**Pull request merged**